File: frontend/src/types/message.ts
export interface AssignedMessageUser {
  id: string;
  fullName?: string;
  email?: string;
  role?: 'student' | 'teacher' | 'admin';
}

export interface MessageRecipient extends AssignedMessageUser {
  fullName: string;
  email: string;
  role: 'student' | 'teacher' | 'admin';
}

export interface SupportMessage {
  id: string;
  name: string;
  email: string;
  phone?: string;
  subject: string;
  content: string;
  summary?: string;
  status: 'unread' | 'read' | 'replied';
  messageType?: 'contact' | 'internal';
  recipientRole?: 'student' | 'teacher' | 'admin';
  readAt?: string;
  repliedAt?: string;
  createdAt?: string;
  assignedTo?: AssignedMessageUser;
  recipientUser?: AssignedMessageUser;
  senderUser?: AssignedMessageUser;
}

File: backend-spring/src/main/java/com/ivyts/backend/domain/message/Message.java
package com.ivyts.backend.domain.message;

import com.ivyts.backend.domain.user.UserRole;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.Locale;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Message {

    private String id;
    private String name;
    private String email;
    private String phone;
    private String subject;
    private String content;
    private String summary;
    private String status = "unread";
    private String messageType = "contact";
    private String recipientRole;
    private String recipientUser;
    private String senderUser;
    private Instant readAt;
    private Instant repliedAt;
    private String assignedTo;
    private Instant createdAt;
    private Instant updatedAt;
    public MessageStatus getStatus() { return MessageStatus.valueOf(normalizeEnum(status, "unread")); }
    public void setStatus(MessageStatus status) { this.status = status.name().toLowerCase(Locale.ROOT); }
    public MessageType getMessageType() { return MessageType.valueOf(normalizeEnum(messageType, "contact")); }
    public void setMessageType(MessageType messageType) { this.messageType = messageType.name().toLowerCase(Locale.ROOT); }
    public UserRole getRecipientRole() { return recipientRole == null ? null : UserRole.valueOf(normalizeEnum(recipientRole, "student")); }
    public void setRecipientRole(UserRole recipientRole) { this.recipientRole = recipientRole == null ? null : recipientRole.name().toLowerCase(Locale.ROOT); }

    private String normalizeEnum(String value, String fallback) {
        return (value == null ? fallback : value).replace('-', '_').toUpperCase(Locale.ROOT);
    }
}
